libsweep/config.mk

```diff
@@ -5,9 +5,9 @@ UNAME := $(shell uname)
 
 ifeq ($(UNAME), Linux)
   PREFIX ?= /usr
-  CFLAGS += -O2 -Wall -Wextra -pedantic -std=c99 -Wnonnull -fvisibility=hidden -fPIC -pthread
+  CXXFLAGS += -O2 -Wall -Wextra -pedantic -std=c++11 -fvisibility=hidden -fPIC -fno-rtti -fno-exceptions -pthread
   LDFLAGS += -shared -Wl,-soname,libsweep.so.$(VERSION_MAJOR)
-  LDLIBS += -lpthread
+  LDLIBS += -lstdc++ -lpthread
```

Comment:
Yes we have to link in the stdlib - at least as soon as we make use of C++ features such as `std::thread` in the implementation.

We could make the compiler and stdlib more easily configurable. On the other hand libstdc++ & gcc is the default on Linux and Clang & libc++ the default on macOS. If users want to change this the `config.mk` file is easily editable.